napari/layers/vectors/vectors.py

```diff
@@ -250,15 +266,13 @@ def _update_thumbnail(self):
         # calculate min vals for the vertices and pad with 0.5
         # the offset is needed to ensure that the top left corner of the
         # vectors corresponds to the top left corner of the thumbnail
-        offset = np.array([self.range[-2][0], self.range[-1][0]]) + 0.5
+        disp = np.where(self.displayed)[0]
+        offset = np.array([self.range[d][0] for d in disp]) + 0.5
```

Comment:
if this is a 1d array
```suggestion
        offset = np.fromiter(self.range[d][0] for d in disp) + 0.5
```

Reply:
fixed